src/connection_check.py

```diff
@@ -35,24 +35,36 @@ def zookeeper_connected(charm: CharmBase) -> bool:
     return True
 
 
+def broker_active(unit: Unit, zookeeper_config: Dict[str, str]) -> bool:
+    """Checks ZooKeeper for client connections, checks for specific broker id.
+
+    Args:
+        unit: the `Unit` to check connection of
+        data: the relation data provided by ZooKeeper
+
+    Returns:
+        True if broker id is recognised as active by ZooKeeper. Otherwise False.
+    """
+    broker_id = unit.name.split("/")[1]
+    return broker_active_internal(broker_id, zookeeper_config)
+
 @retry(
     # retry to give ZK time to update its broker zNodes before failing
     wait=wait_fixed(5),
-    stop=stop_after_attempt(3),
+    stop=stop_after_attempt(20),
```

Comment:
20 here is quite a lot, as I think it would mean that it would be almost 2 minutes before a broker is flagged as non-functional. Could this be lower?

Reply:
Sure Marc, will have to run more tests to tune this.